modules/logging-loki-retention.adoc

```diff
@@ -0,0 +1,76 @@
+// Module included in the following assemblies:
+//
+:_content-type: PROCEDURE
+[id="logging-loki-retention_{context}"]
+= Enabling stream-based retention with Loki
+With Logging version 5.6 and higher, you can configure retention policies based on stream. Rules for these may be set globally, per tenant, or both. If you configure both, tenant rules apply before global rules.
+
+. To enable stream-based retention,  create or edit the `LokiStack` custom resource (CR):
+
+include::snippets/logging-create-apply-cr-snip.adoc[lines=9..12]
+
+.Example LokiStack CR
+
+[source,yaml]
+----
+  apiVersion: loki.grafana.com/v1beta1
+  kind: LokiStack
+  metadata:
+    name: lokistack-dev
+  spec:
+    size: 1x.small
+    storage:
+      secret:
+        name: test
+        type: s3
+    storageClassName: gp2
+    retention: <1>
+      deleteDelay: <2>
+    limits:
+      global: <3>
+        retentionLimits:
+          periodDays: 31
+          stream:
+            - selector:
+                name: namespace
+                type: equal
+                value: dev
+              priority: 1
+              periodDays: 1
+      tenants: <4>
+        tenanta:
+          retentionLimits:
+            periodDays: 7
+            stream:
+              - selector:
+                  name: namespace
+                  type: equal <5>
+                  value: prod
+                priority: 2
+                periodDays: 14
+              - selector:
+                  name: container
+                  type: equal
+                  value: loki
+                priority: 1
+                periodDays: 3
+        tenantb:
+          retentionLimits:
+            periodDays:
+            stream:
+              - selector:
+                  name: container
+                  type: equal
+                  value: nginx
+                priority: 1
+                periodDays: 1
+----
+<1> Retention is enabled in the cluster when this block is added to the CR.
+<2> Time after which the compactor will delete marked chunks.
+<3> Sets retention policy for all logs.
```

Comment:
```suggestion
<3> Sets retention policy for all log streams.
```

We should highlight in a note below that this field is not to be used as the global retention policy of 30days that OpenShift Logging includes as a maximum supported policy. The global 30 days policy across all stored logs has to be defined on the Object Storage provider side for the LokiStack bucket. The key wording here is:
- OpenShift Logging 30 days retention -> Affects **stored logs** on object storage -> Configuration on the object storage provider only
- The field mentioned in `<3>` is a global **log stream** retention policy and applying it is at best effort by the compactor. In addition if someone mixes tenant with global retention policies the global policy is overridden anyway.